### PR TITLE
GCP Label Selector Experiment Charts and GCP Experiments Refactor

### PR DESCRIPTION
**Pull request by neelanjan00**
## Why Do we Need This PR
- Adds GCP VM Instance Stop By Label charts
-  Adds GCP VM Disk Loss By Label charts
- Updates `gcp-vm-disk-loss` and `gcp-vm-instance-stop` charts to include `SEQUENCE` ENV in engine manifest
- Updates `gcp-vm-instance-stop` charts to update `AUTO_SCALING_GROUP` to `MANAGED_INSTANCE_GROUP`
- Updates `gcp-vm-disk-loss` experiment charts to remove `DEVICE_NAMES` ENV

### CODE REVIEW

**Review by avaakash — DISMISSED** [review]

**Comment by avaakash** [comment]
@neelanjan00 can you merge with the base branch.

File: charts/gcp/gcp-vm-disk-loss-by-label/gcp-vm-disk-loss-by-label.chartserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      email: neelanjan.manna@harness.io 
+  minKubeVersion: 1.12.0 
+  provider: 
+    name: Harness
+  labels:
+    app.kubernetes.io/component: chartserviceversion
+    app.kubernetes.io/version: latest
+  links: 
+    - name: Documentation 
+      url: https://docs.litmuschaos.io/docs/getstarted/
```
Comment (Jonsy13) [inline charts/gcp/gcp-vm-disk-loss-by-label/gcp-vm-disk-loss-by-label.chartserviceversion.yaml]: Can we update it to `https://docs.litmuschaos.io/docs/introduction/what-is-litmus` or `https://litmuschaos.github.io/litmus/`

File: charts/gcp/gcp-vm-disk-loss-by-label/experiment.yaml [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        resources: ["secrets","configmaps"]
+        verbs: ["get","list",]
+      # Track and get the runner, experiment, and helper pods log 
+      - apiGroups: [""]
+        resources: ["pods/log"]
+        verbs: ["get","list","watch"]  
+      # for creating and managing to execute comands inside target container
+      - apiGroups: [""]
+        resources: ["pods/exec"]
+        verbs: ["get","list","create"]
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/experiment.yaml]: Can you pls check if we need these permissions?

Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/experiment.yaml]: Can you pls check if we need these permissions?

File: charts/gcp/gcp-vm-disk-loss-by-label/gcp-vm-disk-loss-by-label.chartserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  version: 0.1.0
+  annotations:
+    categories: gcp
+spec:
+  displayName: gcp-vm-disk-loss-by-label 
+  categoryDescription: >
+    gcp vm disk loss by label experiment detaches the target persistent disk volumes by label for a fixed duration of time 
+  keywords: 
+    - "disk" 
+    - "non-kubernetes"
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/gcp-vm-disk-loss-by-label.chartserviceversion.yaml]: we can call it infra

Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/gcp-vm-disk-loss-by-label.chartserviceversion.yaml]: we can call it infra

File: charts/gcp/gcp-vm-disk-loss-by-label/rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    resources: ["secrets","configmaps"]
+    verbs: ["get","list",]
+  # Track and get the runner, experiment, and helper pods log 
+  - apiGroups: [""]
+    resources: ["pods/log"]
+    verbs: ["get","list","watch"]  
+  # for creating and managing to execute comands inside target container
+  - apiGroups: [""]
+    resources: ["pods/exec"]
+    verbs: ["get","list","create"]
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/rbac.yaml:38]: same as above, do we need this?

Comment (uditgaurav) [inline charts/gcp/gcp-vm-disk-loss-by-label/rbac.yaml:38]: same as above, do we need this?

**Review by uditgaurav — APPROVED** [review]

**Review by avaakash — APPROVED** [review]

**Pull request merged**